packages/Checkbox/src/Checkbox.js

```diff
@@ -41,16 +31,19 @@ const Checkbox = props => {
     size,
   };
 
+  if (extendedInputProps.refLabel) {
+    delete extendedInputProps.refLabel;
+  }
+
   const inputProps = {
-    "aria-describedby": ariaDescribedBy,
     checked: checkedState === types.checkboxStates.CHECKED,
     disabled: isDisabled,
-    id: checkboxId,
     onChange,
     ref: inputRef,
     tabIndex,
     type: "checkbox",
     ...extendedInputProps,
+    id: checkboxId,
```

Comment:
we need to make sure the id that the `Checkbox` or `Radio` component generates is used on the input